flask/app/analyses.py

```diff
@@ -244,6 +269,13 @@ def update_analysis(id: int):
         user_id = request.args.get("user")
     else:
         user_id = current_user.user_id
+        if request.json.get("analysis_state") in [
+            "Requested",
+            "Running",
+            "Done",
+            "Error",
+        ]:
```

Comment:
Might it be better to check if this is not Cancelled, since that's the only move authorized for regular users?